index.js

```diff
@@ -337,7 +337,7 @@
                 /* Optimal order of selectors for checking. */
                 var selector_elements = ['year', 'holiday', 'month', 'monthday', 'week', 'weekday'];
                 for (var selector_ind in selector_elements) {
-                    if (selectors[selector_elements[selector_ind]].length > 0) {
+                    if (selector_elements.hasOwnProperty(selector_ind) && selectors[selector_elements[selector_ind]].length > 0) {
```

Comment:
@ypid about formatting... in fact there's probably 2 options:
- eslint (I'm used to use it with [this kind of .eslintrc](https://gist.github.com/shouze/aab54d5f54030cafe091174b18fccb1c) which extends the airbnb one which is pretty complete
- [prettier](https://github.com/prettier/prettier) which is more a bet but very promising I guess, especially as it won't be as complicated as eslint by design for the sake.